lib/charms/operator_libs_linux/v1/snap.py

```diff
@@ -284,7 +280,7 @@ def _install(self, channel: Optional[str] = "", cohort: Optional[str] = "") -> N
         if cohort:
             args.append('--cohort="{}"'.format(cohort))
 
-        self._snap("install", args)
+        self._snap("install", filter(None, args))
```

Comment:
I'd really rather we just built the args cleanly rather than be loose and the fix them immediately after.

Reply:
I'm with you there, was tempted by the low complexity of this solution; instead of burdening each 'should we add this arg' with conditional logic... 
are we sure that would read better?

Reply:
`sed -i ""` is a thing.  So yes - technically possible for meaningful empty args to be used by commands in general.

Reply:
One approach:

```python
args = []
if self._confinement == "classic":
    args.append('--classic')
if channel:
    args.append('--channel')
```

It's more lines, but dead simple.